### ISSUE TITLE
Generic union types don't print the value in FSI

### ISSUE THREAD

**Issue by toburger** [Microsoft/visualfsharp#2438]
When I define or use a generic union type with more than one union case (and one generic case isn't known when requesting the value) the value doesn't get printed out in FSI.

#### Repro steps

```fsharp
Ok "hello"
```

#### Expected behavior

```fsharp
val it : Result<string,'a> = Ok "hello"
```

#### Actual behavior

```fsharp
val it : Result<string,'a>
```

#### Known workarounds

Define all types of the generic parameters, see below.

#### Related information 

This issue does only apply when one of the generic args isn't resolved to an actual type.

The following version of the code does print out the respective value:

```fsharp
> let x: Result<string, string> = Ok "hello";;
val x : Result<string,string> = Ok "hello"
```

This issue does only apply to a multi union case with generic parameters.
The problem doesn't occur with single case union types or with union types with no generic parameters.

This issue is a regression. When using fsi v4.0 the value gets printed.

* Windows 10
* RC3 and master
* .NET Runtime
* Visual Studio 2017 RC4
* fsi.exe, fsiAnyCpu.exe
* Regression

**Comment by dsyme** [comment]
This is actually a general change in behaviour inn F# 4.1 when printing generic values of any kind.  For example:

```
> [];;
val it : 'a list
> (3.14, []);;
val it : float * 'a list
```

This will apply to any value that is not subject to the "value restriction" and is generaized by the F# compiler.  Roughly speaking that means any generic value made up of 
* literal records
* literal union cases
* null
* function values 
* other raw data

The mechanism to print these values was really awkward - basically we instantiated the generics of the value at type ``obj`` and printed that using some really horrid code.  We changed this in FCS somewhere along the way and the change got integrated here.

I'm not certain we should rush to change it back, though I agree the problem is a usability glitch

The problem doesn't apply except at the top-level of F# Interactive, e.g. this is ok:

```
> sprintf "%A" (3.14, []);;
val it : string = "(3.14, [])"
```
